internal/sink/sink.go

```diff
@@ -188,13 +189,28 @@ func (s *Sink) flush() error {
 		offsetStore.Add(message.KafkaMessage.TopicPartition)
 	}
 	offsets := offsetStore.Get()
+	var offsetCommitFailure bool
 	if len(offsets) > 0 {
-		commitedOffsets, err := s.consumer.CommitOffsets(offsets)
+		// We retry with exponential backoff as it's critical that either step #2 or #3 succeeds.
+		err := retry.Do(
```

Comment:
How long do we retry? I don't see that (or any other details about the exponential retry) here.

Reply:
It uses defaults currently:

```
var (
	DefaultAttempts      = [uint](https://pkg.go.dev/builtin#uint)(10)
	DefaultDelay         = 100 * [time](https://pkg.go.dev/time).[Millisecond](https://pkg.go.dev/time#Millisecond)
	DefaultMaxJitter     = 100 * [time](https://pkg.go.dev/time).[Millisecond](https://pkg.go.dev/time#Millisecond)
	DefaultOnRetry       = func(n [uint](https://pkg.go.dev/builtin#uint), err [error](https://pkg.go.dev/builtin#error)) {}
	DefaultRetryIf       = [IsRecoverable](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#IsRecoverable)
	DefaultDelayType     = [CombineDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#CombineDelay)([BackOffDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#BackOffDelay), [RandomDelay](https://pkg.go.dev/github.com/avast/retry-go?utm_source=godoc#RandomDelay))
	DefaultLastErrorOnly = [false](https://pkg.go.dev/builtin#false)
	DefaultContext       = [context](https://pkg.go.dev/context).[Background](https://pkg.go.dev/context#Background)()
)
```

Reply:
It's a backoff with jitter